docs/extensibility/custom-resources.md

```diff
@@ -0,0 +1,327 @@
+```
+
+Inside the `Program.cs` in the `MailDevResource.AppHost` project, update the source file to look like the following:
+
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
+var maildev = builder.AddMailDev("maildev");
+builder.AddProject<Projects.MailDevResource_NewsletterService>("newsletterservice")
+       .WithReference(maildev);
+
+builder.Build().Run();
+```
+
+After updating the `Program.cs` source file launch the AppHost again and verify that the .NET project that was added started and that the environment variable `ConnectionStrings__maildev` was added to the process (click on the _Details_ link on the `newsletterservice` row in the dashboard).
+
+![Environment variables for Newsletter Service in .NET Aspire Dashboard](./media/maildev-envvar.png)
+
+### Step 4b: Use connection string to send message.
+
+To use the SMTP connection details that were injected into the newsletter service project we will inject an instance of <xref:System.Net.Mail.SmtpClient> into the dependency injection container as a singleton. Add the following code to the `Program.cs` file in the `MailDevResource.NewsletterService` project to setup the singleton service.
+
+```csharp
+builder.Services.AddSingleton<SmtpClient>((sp) =>
+{
+    var smtpUri = new Uri(builder.Configuration.GetConnectionString("maildev")!);
+    var smtpClient = new SmtpClient(smtpUri.Host, smtpUri.Port);
+    return smtpClient;
+});
+```
+
+To test the client we will add two simple subscribe and unsubscribe GET methods to the newsletter service. Add the following code to the `Program.cs` file in the `MailDevResource.NewsletterService` project to setup the ASP.NET Core routes:
+
+```csharp
+app.MapGet("/subscribe", async (SmtpClient smtpClient, string email) => {
+    var message = new MailMessage("newsletter@yourcompany.com", email);
+    message.Subject = "Welcome to our newsletter!";
+    message.Body = "Thank you for subscribing to our newsletter!";
+    await smtpClient.SendMailAsync(message);
+});
+
+app.MapGet("/unsubscribe", async (SmtpClient smtpClient, string email) => {
+    var message = new MailMessage("newsletter@yourcompany.com", email);
+    message.Subject = "You are unsubscribed from our newsletter!";
+    message.Body = "Sorry to see you go. We hope you will come back soon!";
+    await smtpClient.SendMailAsync(message);
+});
+```
+
+Once the `Program.cs` file is updated launch the AppHost and use your browser, or `curl` to hit the following URLs:
+
+```bash
+# Remember to replace the port to match your local environment.
+curl https://localhost:7289/subscribe?email=test@test.com
+curl https://localhost:7289/unsubscribe?email=test@test.com
+```
+
+If those API calls return a successful response then you should be able to click on the `maildev` resource the dashboard and the MailDev UI will show the emails that have been sent to the SMTP endpoint.
+
+![E-mails visible in the MailDev UI](./media/maildev-emails.png)
```

Comment:
I think we need an explanation of what we did and the pieces after the sample or maybe explained after each section. It would also be nice to show the manifest representation of this resource.

Reply:
Expanded the technical details. Let me know if this covers what you are after. If we want more I think it needs to be a separate article because it can be pretty overwhelming.

Reply:
It looks good.